lib/src/git.rs

```diff
@@ -928,14 +928,20 @@ pub fn reset_head(
     wc_commit: &Commit,
 ) -> Result<(), git2::Error> {
     let first_parent_id = &wc_commit.parent_ids()[0];
+    let first_parent;
```

Comment:
nit: I slightly prefer this:
```rust
    let first_parent = if first_parent_id != mut_repo.store().root_commit_id() {
        RefTarget::normal(first_parent_id.clone())
    } else {
        RefTarget::absent()
    };
    if first_parent.is_present() {
```

`mut_repo.git_head().is_present()` in `else` clause is equivalent to `git_head != &first_parent`, and we might reorganize them later.

Reply:
Done.